### ISSUE TITLE
Fail to launch Triton-Server with Whisper TRT-LLM

### ISSUE THREAD

**Issue by danhayoun** [triton-inference-server/tensorrtllm_backend#835]
### System Info

- Architecture : x86_64
- OS : Ubuntu 24.04 LTS
- GPU : NVIDIA  Geforce RTX 3090 (compute capability 8.6) et NVIDIA RTX 2080 Ti (compute capability 7.5)
- Python : 3.12.3
- Driver NVIDIA : 575.57.08

**Context**

I'm trying to reproduce exactly the guide shown here : https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/docs/whisper.md 

I successfully converted the checkpoints and build the engines. But the Triton Server fails to load.

I do everything in this docker image : nvcr.io/nvidia/tritonserver:25.06-trtllm-python-py3 so in version 25.06.

### Who can help?

@ncomly-nvidia 

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction
`
git clone https://github.com/triton-inference-server/tensorrtllm_backend.git && cd tensorrtllm_backend
git lfs install
git submodule update --init --recursive

docker run --rm -ti --net=host -v `pwd`:/mnt -w /mnt --gpus all nvcr.io/nvidia/tritonserver:25.06-trtllm-python-py3 bash

cd tensorrt_llm/examples/models/core/whisper/

wget --directory-prefix=assets https://raw.githubusercontent.com/openai/whisper/main/whisper/assets/multilingual.tiktoken
wget --directory-prefix=assets assets/mel_filters.npz https://raw.githubusercontent.com/openai/whisper/main/whisper/assets/mel_filters.npz
wget --directory-prefix=assets https://raw.githubusercontent.com/yuekaizhang/Triton-ASR-Client/main/datasets/mini_en/wav/1221-135766-0002.wav
# take large-v3 model as an example
wget --directory-prefix=assets https://openaipublic.azureedge.net/main/whisper/models/e5b1a55b89c1367dacf97e3e19bfd829a01529dbfdeefa8caeb59b3f1b81dadb/large-v3.pt

INFERENCE_PRECISION=float16
MAX_BEAM_WIDTH=4
MAX_BATCH_SIZE=64
checkpoint_dir=tllm_checkpoint
output_dir=whisper_large_v3_max_batch_${MAX_BATCH_SIZE}
MODEL_DIR=assets

python3 convert_checkpoint.py --model_dir ${MODEL_DIR} --output_dir ${checkpoint_dir}

trtllm-build --checkpoint_dir ${checkpoint_dir}/encoder \
            --output_dir ${output_dir}/encoder \
            --moe_plugin disable \
            --max_batch_size ${MAX_BATCH_SIZE} \
            --gemm_plugin disable \
            --bert_attention_plugin ${INFERENCE_PRECISION} \
            --max_input_len 3000 --max_seq_len=3000

trtllm-build  --checkpoint_dir ${checkpoint_dir}/decoder \
            --output_dir ${output_dir}/decoder \
            --moe_plugin disable \
            --max_beam_width ${MAX_BEAM_WIDTH} \
            --max_batch_size ${MAX_BATCH_SIZE} \
            --max_seq_len 114 \
            --max_input_len 14 \
            --max_encoder_input_len 3000 \
            --gemm_plugin ${INFERENCE_PRECISION} \
            --bert_attention_plugin ${INFERENCE_PRECISION} \
            --gpt_attention_plugin ${INFERENCE_PRECISION}

cd /mnt

cp tensorrt_llm/triton_backend/all_models/whisper/ model_repo_whisper -r
cp tensorrt_llm/triton_backend/all_models/inflight_batcher_llm/tensorrt_llm model_repo_whisper -r
wget --directory-prefix=model_repo_whisper/whisper_bls/1 https://raw.githubusercontent.com/openai/whisper/main/whisper/assets/multilingual.tiktoken
wget --directory-prefix=model_repo_whisper/whisper_bls/1 https://raw.githubusercontent.com/openai/whisper/main/whisper/assets/mel_filters.npz

BACKEND=tensorrtllm
DECOUPLED_MODE=false
DECODER_ENGINE_PATH=${output_dir}/decoder
ENCODER_ENGINE_PATH=${output_dir}/encoder
MAX_TOKENS_IN_KV_CACHE=24000
BATCHING_STRATEGY=inflight_fused_batching
KV_CACHE_FREE_GPU_MEM_FRACTION=0.5
EXCLUDE_INPUT_IN_OUTPUT=True
TRITON_MAX_BATCH_SIZE=8
MAX_QUEUE_DELAY_MICROSECONDS=0
MAX_BEAM_WIDTH=1
MAX_QUEUE_SIZE="0"
ENABLE_KV_CACHE_REUSE=false
ENABLE_CHUNKED_CONTEXT=false
CROSS_KV_CACHE_FRACTION="0.5"
n_mels=128
zero_pad=false

python3 tensorrt_llm/triton_backend/tools/fill_template.py -i model_repo_whisper/tensorrt_llm/config.pbtxt triton_backend:${BACKEND},engine_dir:${DECODER_ENGINE_PATH},encoder_engine_dir:${ENCODER_ENGINE_PATH},decoupled_mode:${DECOUPLED_MODE},max_tokens_in_paged_kv_cache:${MAX_TOKENS_IN_KV_CACHE},max_attention_window_size:${MAX_ATTENTION_WINDOW_SIZE},batch_scheduler_policy:${BATCH_SCHEDULER_POLICY},batching_strategy:${BATCHING_STRATEGY},kv_cache_free_gpu_mem_fraction:${KV_CACHE_FREE_GPU_MEM_FRACTION},exclude_input_in_output:${EXCLUDE_INPUT_IN_OUTPUT},triton_max_batch_size:${TRITON_MAX_BATCH_SIZE},max_queue_delay_microseconds:${MAX_QUEUE_DELAY_MICROSECONDS},max_beam_width:${MAX_BEAM_WIDTH},enable_kv_cache_reuse:${ENABLE_KV_CACHE_REUSE},normalize_log_probs:${NORMALIZE_LOG_PROBS},enable_chunked_context:${ENABLE_CHUNKED_CONTEXT},gpu_device_ids:${GPU_DEVICE_IDS},decoding_mode:${DECODING_MODE},max_queue_size:${MAX_QUEUE_SIZE},enable_context_fmha_fp32_acc:${ENABLE_CONTEXT_FMHA_FP32_ACC},cross_kv_cache_fraction:${CROSS_KV_CACHE_FRACTION},encoder_input_features_data_type:TYPE_FP16,logits_datatype:TYPE_FP32,prompt_embedding_table_data_type:TYPE_FP16

python3 tensorrt_llm/triton_backend/tools/fill_template.py -i model_repo_whisper/whisper_bls/config.pbtxt engine_dir:${ENCODER_ENGINE_PATH},n_mels:$n_mels,zero_pad:$zero_pad,triton_max_batch_size:${TRITON_MAX_BATCH_SIZE},decoupled_mode:${DECOUPLED_MODE}

python3 tensorrt_llm/triton_backend/scripts/launch_triton_server.py --world_size 1 --model_repo=model_repo_whisper/ --tensorrt_llm_model_name tensorrt_llm,whisper_bls --multimodal_gpu0_cuda_mem_pool_bytes 300000000

``
### Expected behavior

Triton Server successfully initialized.

### actual behavior

The Triton Server fails to initialize with this error : 
`
backend_model.cc:694] "ERROR: Failed to create instance: unexpected error when creating modelInstanceState: [TensorRT-LLM][ERROR] Assertion failed: Both encoder and decoder model paths are empty (/tmp/tritonbuild/tensorrtllm/triton_backend/inflight_batcher_llm/src/model_instance_state.cc:771)\n1       0x7f27103bd242 /opt/tritonserver/backends/tensorrtllm/libtriton_tensorrtllm.so(+0x17242) [0x7f27103bd242]\n2       0x7f27103bcea2 /opt/tritonserver/backends/tensorrtllm/libtriton_tensorrtllm.so(+0x16ea2) [0x7f27103bcea2]\n3       0x7f27103de292 triton::backend::inflight_batcher_llm::ModelInstanceState::Create(triton::backend::inflight_batcher_llm::ModelState*, TRITONBACKEND_ModelInstance*, triton::backend::inflight_batcher_llm::ModelInstanceState**) + 66\n4       0x7f27103c7b09 TRITONBACKEND_ModelInstanceInitialize + 153\n5       0x7f272e9d6799 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x1a1799) [0x7f272e9d6799]\n6       0x7f272e9d7222 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x1a2222) [0x7f272e9d7222]\n7       0x7f272e9bce43 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x187e43) [0x7f272e9bce43]\n8       0x7f272e9bd1f4 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x1881f4) [0x7f272e9bd1f4]\n9       0x7f272e9c67dd /opt/tritonserver/bin/../lib/libtritonserver.so(+0x1917dd) [0x7f272e9c67dd]\n10      0x7f272df33ed3 /usr/lib/x86_64-linux-gnu/libc.so.6(+0xa1ed3) [0x7f272df33ed3]\n11      0x7f272e9b4052 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x17f052) [0x7f272e9b4052]\n12      0x7f272e9c1f2c /opt/tritonserver/bin/../lib/libtritonserver.so(+0x18cf2c) [0x7f272e9c1f2c]\n13      0x7f272e9c5f62 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x190f62) [0x7f272e9c5f62]\n14      0x7f272eac75e1 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x2925e1) [0x7f272eac75e1]\n15      0x7f272eaca93c /opt/tritonserver/bin/../lib/libtritonserver.so(+0x29593c) [0x7f272eaca93c]\n16      0x7f272ec26e85 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x3f1e85) [0x7f272ec26e85]\n17      0x7f272e190db4 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0xecdb4) [0x7f272e190db4]\n18      0x7f272df2eaa4 /usr/lib/x86_64-linux-gnu/libc.so.6(+0x9caa4) [0x7f272df2eaa4]\n19      0x7f272dfbba34 __clone + 68"
E0109 15:59:30.921806 138 model_lifecycle.cc:654] "failed to load 'tensorrt_llm' version 1: Internal: unexpected error when creating modelInstanceState: [TensorRT-LLM][ERROR] Assertion failed: Both encoder and decoder model paths are empty (/tmp/tritonbuild/tensorrtllm/triton_backend/inflight_batcher_llm/src/model_instance_state.cc:771)\n1       0x7f27103bd242 /opt/tritonserver/backends/tensorrtllm/libtriton_tensorrtllm.so(+0x17242) [0x7f27103bd242]\n2       0x7f27103bcea2 /opt/tritonserver/backends/tensorrtllm/libtriton_tensorrtllm.so(+0x16ea2) [0x7f27103bcea2]\n3       0x7f27103de292 triton::backend::inflight_batcher_llm::ModelInstanceState::Create(triton::backend::inflight_batcher_llm::ModelState*, TRITONBACKEND_ModelInstance*, triton::backend::inflight_batcher_llm::ModelInstanceState**) + 66\n4       0x7f27103c7b09 TRITONBACKEND_ModelInstanceInitialize + 153\n5       0x7f272e9d6799 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x1a1799) [0x7f272e9d6799]\n6       0x7f272e9d7222 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x1a2222) [0x7f272e9d7222]\n7       0x7f272e9bce43 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x187e43) [0x7f272e9bce43]\n8       0x7f272e9bd1f4 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x1881f4) [0x7f272e9bd1f4]\n9       0x7f272e9c67dd /opt/tritonserver/bin/../lib/libtritonserver.so(+0x1917dd) [0x7f272e9c67dd]\n10      0x7f272df33ed3 /usr/lib/x86_64-linux-gnu/libc.so.6(+0xa1ed3) [0x7f272df33ed3]\n11      0x7f272e9b4052 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x17f052) [0x7f272e9b4052]\n12      0x7f272e9c1f2c /opt/tritonserver/bin/../lib/libtritonserver.so(+0x18cf2c) [0x7f272e9c1f2c]\n13      0x7f272e9c5f62 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x190f62) [0x7f272e9c5f62]\n14      0x7f272eac75e1 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x2925e1) [0x7f272eac75e1]\n15      0x7f272eaca93c /opt/tritonserver/bin/../lib/libtritonserver.so(+0x29593c) [0x7f272eaca93c]\n16      0x7f272ec26e85 /opt/tritonserver/bin/../lib/libtritonserver.so(+0x3f1e85) [0x7f272ec26e85]\n17      0x7f272e190db4 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0xecdb4) [0x7f272e190db4]\n18      0x7f272df2eaa4 /usr/lib/x86_64-linux-gnu/libc.so.6(+0x9caa4) [0x7f272df2eaa4]\n19      0x7f272dfbba34 __clone + 68"
``

I haven't seen this error documented elsewhere and was unable to debug it.

### additional notes

I also tried to follow the guide with version 24.07 of the image (so nvcr.io/nvidia/tritonserver:24.07-trtllm-python-py3) and it failed with the same mistake. My ultimate goal is to implement whisper, with Triton tensorrtllm, as one model in my Triton Server. Then I'll be able to send requests to it from different models and have it run with it's own tensorrtllm backend (not python). 

How can I solve this ?
Thank you
